x-pack/plugins/security/public/nav_control/nav_control_component.tsx

```diff
@@ -93,17 +92,20 @@ export const SecurityNavControl: FunctionComponent<SecurityNavControlProps> = ({
     items.push(...userMenuLinkMenuItems);
   }
 
-  if (!isAnonymous) {
+  const isAnonymous = currentUser.value ? isUserAnonymous(currentUser.value) : false;
+  const isCloudUser = currentUser.value?.elastic_cloud_user;
+
+  if (!isAnonymous && !isCloudUser) {
     const hasCustomProfileLinks = userMenuLinks.some(({ setAsProfile }) => setAsProfile === true);
+
     const profileMenuItem: EuiContextMenuPanelItemDescriptor = {
       name: (
         <FormattedMessage
           id="xpack.security.navControlComponent.editProfileLinkText"
-          defaultMessage="{profileOverridden, select, true{Preferences} other{Profile}}"
-          values={{ profileOverridden: hasCustomProfileLinks }}
+          defaultMessage="Edit Profile"
```

Comment:
__nit:__ I think you also need to update text here https://github.com/elastic/kibana/blob/754fdea25de76006c0a26d8aeda87bdd0a426dd2/x-pack/plugins/cloud/public/user_menu_links.ts#L20 to have the same text for all users (either Cloud managed or not).

Reply:
Resolved in latest commit!